// ocrmypdf-js/dist/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var __async = (__this, __arguments, generator) => {
  return new Promise((resolve, reject) => {
    var fulfilled = (value) => {
      try {
        step(generator.next(value));
      } catch (e) {
        reject(e);
      }
    };
    var rejected = (value) => {
      try {
        step(generator.throw(value));
      } catch (e) {
        reject(e);
      }
    };
    var step = (x) => x.done ? resolve(x.value) : Promise.resolve(x.value).then(fulfilled, rejected);
    step((generator = generator.apply(__this, __arguments)).next());
  });
};

// src/index.ts
var src_exports = {};
__export(src_exports, {
  OcrMyPdf: () => OcrMyPdf
});
module.exports = __toCommonJS(src_exports);

// src/utils/execShellCommand.ts
var import_node_child_process = require("child_process");
function execShellCommand(command) {
  return __async(this, null, function* () {
    return new Promise((resolve, reject) => {
      (0, import_node_child_process.exec)(command, (err, stdout, sterr) => {
        if (err) {
          return reject(err);
        }
        return resolve(stdout || sterr);
      });
    });
  });
}

// src/utils/logger.ts
var import_cli_color = __toESM(require("cli-color"));
var logger = {
  info: (message) => {
    const date = (/* @__PURE__ */ new Date()).toLocaleTimeString();
    console.log(import_cli_color.default.white(`[${date}] `) + import_cli_color.default.green(message));
  },
  warn: (message) => {
    const date = (/* @__PURE__ */ new Date()).toLocaleTimeString();
    console.log(import_cli_color.default.yellow(`[${date}] ${message}`));
  },
  error: (message, locale) => {
    const date = (/* @__PURE__ */ new Date()).toLocaleTimeString();
    console.info(
      `${import_cli_color.default.white(`[${date}] `)} ${import_cli_color.default.red(message)} origin:${import_cli_color.default.white(
        locale || ""
      )}`
    );
  }
};

// src/lib/OcrMyPdf.ts
var OcrMyPdf = class {
  /**
   * Creates an instance of the OcrMyPdf class.
   * @param params  - Optional parameters for configuring the OcrMyPdf instance.
   */
  constructor(params) {
    Object.assign(this, params);
  }
  /**
   * Executes OCR on the PDF file.
   * @param params - Optional parameters for executing OCR.
   * @throws Throws an error if the execution fails.
   */
  execute(params) {
    return __async(this, null, function* () {
      logger.info("Executing OCR on the PDF file...");
      try {
        return yield this.executeOcrMyPdfInShell(params);
      } catch (error) {
        logger.error(
          `Failed to execute OCR on the PDF file: ${error.message}`,
          this.execute.name
        );
        throw error;
      } finally {
        logger.info("Finished executing OCR on the PDF file.");
      }
    });
  }
  /**
   * Executes a raw command in the shell.
   * @param args - The command to execute.
   * @throws Throws an error if the execution fails.
   * @returns The output of the command.
   */
  executeRaw(args) {
    return __async(this, null, function* () {
      logger.info("Executing a raw command in the shell...");
      try {
        return yield execShellCommand(args);
      } catch (error) {
        logger.error(
          `Failed to execute a raw command in the shell: ${error.message}`
        );
        throw error;
      } finally {
        logger.info("Finished executing a raw command in the shell.");
      }
    });
  }
  executeOcrMyPdfInShell(params) {
    return __async(this, null, function* () {
      try {
        const args = [].concat((params == null ? void 0 : params.args) || "").concat((this == null ? void 0 : this.args) || "");
        const inputPath = (params == null ? void 0 : params.inputPath) || this.inputPath;
        const outputPath = (params == null ? void 0 : params.outputPath) || this.outputPath;
        if (!inputPath || !outputPath)
          throw new Error("inputPath or outputPath is not defined!");
        yield execShellCommand(
          `ocrmypdf ${args == null ? void 0 : args.join(" ")} ${inputPath} ${outputPath}`
        );
        return {
          outputPath
        };
      } catch (error) {
        throw error;
      }
    });
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  OcrMyPdf
});
